js/src/components/help-popover/index.js

```diff
@@ -20,30 +22,45 @@ import './index.scss';
 
 /**
  * @param {Object} props React props
- * @param {string} props.id The Popover’s ID
+ * @param {string} [props.className] Additional CSS class name to be appended.
+ * @param {string} [props.id] The Popover’s ID for event tracking.
+ * @param {boolean} [props.disabled=false] Whether to disable the help icon button and also hide the popover.
+ * @param {number} [props.iconSize=16] Size of the help icon.
  * @param {Array<JSX.Element>} props.children The Popover’s content
  * @fires gla_tooltip_viewed with the given `id`.
  */
-const HelpPopover = ( { id, children, ...props } ) => {
+const HelpPopover = ( {
+	className,
+	id,
+	disabled = false,
+	iconSize = 16,
+	children,
+	...props
+} ) => {
 	const [ showPopover, setShowPopover ] = useState( false );
 
 	const handleButtonClick = () => {
 		setShowPopover( true );
-		recordEvent( 'gla_tooltip_viewed', {
-			id,
-		} );
+
+		if ( id ) {
+			recordEvent( 'gla_tooltip_viewed', { id } );
+		}
```

Comment:
❓  Why are we making this optional now? It's a requirement ?

Reply:
Whether to do event tracking here is not yet decided. With the current implementation, `AssetField` uses `HelpPopover` but doesn't set `id` prop, so it was changed to optional.